synapse/models/dns.py

```diff
@@ -37,73 +37,163 @@ def getBaseModels():
                     'fields': 'fqdn,inet:fqdn|ipv6,inet:ipv6',
                     'doc': 'The result of a DNS AAAA record lookup',
                     'ex': 'vertex.link/2607:f8b0:4004:809::200e'}),
+
+                ('inet:dns:cname', {
+                     'subof': 'sepr',
+                     'sep': '/',
+                     'fields': 'fqdn,inet:fqdn|cname,inet:fqdn',
+                     'doc': 'The result of a DNS CNAME record lookup',
+                     'ex': 'foo.vertex.link/vertex.link'}),
+
+                ('inet:dns:mx', {
+                     'subof': 'sepr',
+                     'sep': '/',
+                     'fields': 'fqdn,inet:fqdn|mx,inet:fqdn',
+                     'doc': 'The result of a DNS MX record lookup',
+                     'ex': 'vertex.link/mail.vertex.link'}),
+
+                #An SOA record may return a domain or email address or both. RFC1035.
+
+                ('inet:dns:soa', {
```

Comment:
SOA gets a bit sticky...  the "rname" field in the SOA response may be a domain, an email address, or in practice basically any string ( with no way other than "does this smell like an inet:fqdn/inet:email string?" to tell the diff. )  Without a way to distinguish, it falls to a parser to "decide" which kind (if any) it is.  Since it can be nearly any free-form string, it may be better to just store it as is in a single field.

Also, unlike most of the other RR types, an SOA lookup returns a few other things that we may want to include in making something "unique" ( specifically mname/serial.  But especially serial since, in practice with BIND, they tend to change every time someone makes a change to any record within the zone ).  These could also be extra fields in a per-instance protocol parser representation, but they seem worth considering.  Thoughts?